x/interchainstaking/ibc_module.go

```diff
@@ -217,6 +218,14 @@ func (im IBCModule) OnAcknowledgementPacket(
 	acknowledgement []byte,
 	relayer sdk.AccAddress,
 ) error {
+	connectionId, _, err := im.keeper.IBCKeeper.ChannelKeeper.GetChannelConnection(ctx, packet.SourcePort, packet.SourceChannel)
```

Comment:
is `packet` here the original packet being acknowledged (i.e. source is our local channel and port) - I think it is (in which case this is fine!).

Reply:
I am a little confused here, but as you mention it seems to be the original packet, thus the source would refer to the local chain channel and port;

OnAcknowledgementPacket signature suggests packet is the original accompanied by acknowledgement []byte;